### DISCUSSION TITLE
syntax error, unexpected ')', expecting variable (T_VARIABLE)

### DISCUSSION THREAD

**Discussion by kb6673** [spatie/laravel-medialibrary#2431] [Q&A] [answered]
I've got a fairly simple model setup

```
  public function registerMediaCollections(): void {
        $this->addMediaCollection( 'logo' )->singleFile();
    }
```

With a single file uploader like this

```
   <div class="mt-2">
            <label class="mb-1">Logo</label>
            <x-media-library-attachment name="logo" rules="mimes:png,jpg" />
        </div>
```

Controller

```
    public function store( CompanyRequest $request ) {
        $validated = $request->validated();

        $company                 = new Company;
...
        $company->save();

        if ( isset( $validated['logo'] ) ) {
            $company->syncFromMediaLibraryRequest( $validated['logo'] )->toMediaCollection( 'logo' );
        }

        return redirect( route( 'companies.show', $company->slug ) )->with( 'success', 'Successfully created new company - ' . $company->name );
    }
```

Request

```
use Spatie\MediaLibraryPro\Rules\Concerns\ValidatesMedia;

class CompanyRequest extends FormRequest {

    use ValidatesMedia;

    public function rules() {
        return [
            ...
            'logo'           => [
                'nullable',
                $this->validateSingleMedia()->extension( [ 'png', 'jpg' ] )
            ]
        ];
    }
```

Whenever I fill in the file upload and submit I get `syntax error, unexpected ')', expecting variable (T_VARIABLE) `

![image](https://user-images.githubusercontent.com/1001484/117433549-7ebd8d80-af23-11eb-904e-ca24cff5b3e5.png)

Dump of the file object

![image](https://user-images.githubusercontent.com/1001484/117436987-82531380-af27-11eb-910a-aea6db0b6274.png)

Same error if I bypass the Request too.

Same error if I use `addFromMediaLibraryRequest`

Same error if I use Blade or Livewire

I am clearly doing something wrong and I can't see what I've done compared to the demo app.

Can anyone spot the obvious?

**Comment by freekmurze** [comment, marked as answer]
This has been fixed in media-library-pro. Run `composer update` to pull in the fix.

**Reply by kb6673** [reply]
That's perfect. Thank you
